### PR TITLE
[ADD] Change GGN to purely PyTorch

### PR DESCRIPTION
**Pull request by f-dangel**
Only review after #142, #145.

### CODE REVIEW

File: curvlinops/ggn.py [hunk 1]
```diff
@@ -39,47 +39,42 @@ class GGNLinearOperator(_LinearOperator):
...
+            M: Matrix to be multiplied with in tensor list format.
                 Tensors have same shape as trainable model parameters, and an
-                additional leading axis for the matrix columns.
+                additional trailing axis for the matrix columns.
 
         Returns:
             Result of GGN multiplication in list format. Has the same shape as
-            ``M_list``, i.e. each tensor in the list has the shape of a parameter and a
-            leading dimension of matrix columns.
+            ``M_``, i.e. each tensor in the list has the shape of a parameter and a
```
Comment (runame) [inline curvlinops/ggn.py:63]: `M` instead of `M_`?

File: docs/examples/basic_usage/example_fisher_monte_carlo.py [hunk 2]
```diff
@@ -78,7 +78,7 @@
 # Fisher and compute their matrix representations by multiplying them onto the
 # identity matrix:
 
-GGN = GGNLinearOperator(model, loss_function, params, data)
+GGN = GGNLinearOperator(model, loss_function, params, data).to_scipy()
```
Comment (runame) [inline docs/examples/basic_usage/example_fisher_monte_carlo.py:81]: What is the reason to call `.scipy()` for `GGN` but not for `F`?

**Review by runame — APPROVED** [review]
LGTM!

File: curvlinops/ggn.py (hunk 1 above)
Comment (f-dangel) [reply curvlinops/ggn.py:63]: Good catch.

File: docs/examples/basic_usage/example_fisher_monte_carlo.py (hunk 2 above)
Comment (f-dangel) [reply docs/examples/basic_usage/example_fisher_monte_carlo.py:81]: This PR only changes the GGN linear operator to pure PyTorch. The Fisher linear operator is still a normal SciPy linear operator, therefore it does not need to be converted explicitly.

Comment (runame) [reply docs/examples/basic_usage/example_fisher_monte_carlo.py:81]: ahh of course, thanks!

**Pull request merged**